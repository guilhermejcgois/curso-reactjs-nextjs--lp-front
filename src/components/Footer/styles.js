import style, { css } from 'styled-components';

import { Container as SectionContainer } from '../SectionContainer/styles';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = style.p`
  ${({ theme }) => css`
    text-align: center;
    border-top: 0.1rem solid ${theme.colors.mediumGray};

    a {
      color: inherit;
      text-decoration: none;
    }

    & ${SectionContainer} {
      padding-top: 0;
      padding-bottom: 0;
    }

    & ${TextComponent} {
      font-size: ${theme.font.sizes.small};
    }
  `}
`;
