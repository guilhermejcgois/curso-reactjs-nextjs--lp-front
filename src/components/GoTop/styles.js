import styled from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => `
    position: fixed;
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    display: flex;
    align-items: center;
    justify-contener: center;
    width: 4rem;
    height: 4rem;
    bottom: 2rem;
    right: 2rem;
    z-index: 6;
  `}
`;
