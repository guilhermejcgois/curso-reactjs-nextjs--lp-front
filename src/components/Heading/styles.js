import styled, { css } from 'styled-components';

const titleSize = {
  small: (theme) =>
    css`
      font-size: ${theme.font.sizes.medium};
    `,
  medium: (theme) =>
    css`
      font-size: ${theme.font.sizes.large};
    `,
  big: (theme) =>
    css`
      font-size: ${theme.font.sizes.huge};
    `,
  huge: (theme) =>
    css`
      font-size: ${theme.font.sizes.xhuge};
    `,
};

const titleCase = (uppercase) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1`
  ${({ theme, colorDark, size, uppercase }) => css`
    color: ${colorDark ? theme.colors.primaryColor : theme.colors.light};
    ${titleSize[size](theme)}
    ${titleCase(uppercase)}
  `}
`;
