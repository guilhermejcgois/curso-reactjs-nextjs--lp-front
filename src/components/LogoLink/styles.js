import style, { css } from 'styled-components';

export const Container = style.a`
  ${() => css`
    display: flex;
    align-items: center;
    text-decoration: none;
    color: inherit;

    > img {
      height: 3rem;
    }
  `}
`;
