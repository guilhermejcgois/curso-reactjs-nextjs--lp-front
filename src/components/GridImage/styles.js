import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    ${TextComponent} {
      margin-bottom: ${theme.spacing.xhuge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacing.large};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  overflow: hidden;
`;

export const Image = styled.img`
  width: 100%;
  transition: all 300ms ease-in-out;

  &:hover {
    transform: scale(1.2) rotate(10deg);
  }
`;
