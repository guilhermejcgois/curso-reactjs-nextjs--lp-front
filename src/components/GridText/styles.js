import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    ${TextComponent} {
      margin-bottom: ${theme.spacing.xhuge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacing.large};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    ${HeadingContainer} {
      position: relative;
      left: 5rem;
    }

    ${HeadingContainer}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position absolute;
      font-size: ${theme.font.sizes.xhuge};
      top -3rem;
      left: -5rem;
      transform: rotate(5deg);
    }
  `}
`;
