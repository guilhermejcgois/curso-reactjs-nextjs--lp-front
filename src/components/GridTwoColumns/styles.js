import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 1fr 2fr;
    align-items: center;
    gap: ${theme.spacing.large};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
      text-align: center;
    }

    ${Title} {
      margin-bottom: ${theme.spacing.xlarge};
    }
  `}
`;

export const ImageContainer = styled.div``;

export const Image = styled.img``;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    @media ${theme.media.lteMedium} {
      margin-bottom: ${theme.spacing.large};
    }
  `}
`;
