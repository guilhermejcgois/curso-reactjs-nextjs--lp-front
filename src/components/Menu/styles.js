import styled, { css } from 'styled-components';
import { Title as Heading } from '../Heading/styles';
import { Container as SectionContainer } from '../SectionContainer/styles';

const menuVisible = () => css`
  visibility: visible;
  opacity: 1;
`;

export const Container = styled.div`
  ${({ visible, theme }) => css`
    position: 0;
    z-index: 5;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    width: 100%;
    height: 100%;
    border-bottom: ${theme.colors.mediumGray};
    transition: all 300ms ease-in-out;

    > ${SectionContainer} {
      padding-top: 0;
      padding-bottom: 0;
    }

    & ${Heading} {
      margin-top: 0;
      margin-bottom: 0;
    }

    @media ${theme.media.lteMedium} {
      height: 100vh;
      visibility: hidden;
      opacity: 0;
      ${visible && menuVisible()}

      > ${SectionContainer} {
        display: grid;
        grid-template-columns: 1fr;
        grid-template-rows: 1fr;
        height: 100vh;
        align-items: center;
        overflow-y: auto;
      }

      & ${Heading} {
        padding-bottom: ${theme.spacing.large};
        display: flex;
        justify-content: center;
      }
    }
  `}
`;

export const MenuContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: space-between;
    align-items: center;

    @media ${theme.media.lteMedium} {
      display: block;
      text-align: center;
      padding: ${theme.spacing.xxlarge};
    }
  `}
`;

export const Button = styled.button`
  ${({ visible, theme }) => css`
    display: none;
    z-index: 6;
    position: fixed;
    top: 2rem;
    right: 2rem;
    width: 4rem;
    height: 4rem;
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    border: none;
    pointer-events: ${visible ? 'none' : 'all'};

    @media ${theme.media.lteMedium} {
      display: flex;
      align-items: center;
      justify-content: center;
    }

    > svg {
      width: 2.5rem;
      height: 2.5rem;
    }
  `}
`;
