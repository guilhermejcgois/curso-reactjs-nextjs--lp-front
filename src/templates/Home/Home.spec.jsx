import React from 'react';
import { screen } from '@testing-library/react';

import { Home } from '.';
import { renderTheme } from '../../styles/render-theme';
import { theme } from '../../styles/theme';

describe('<Home />', () => {
  it('should render', () => {
    renderTheme(<Home />);
    const headingContainer = screen.getByRole('heading', { name: 'Hello' }).parentElement;

    expect(headingContainer).toHaveStyle({
      background: theme.colors.mainBg,
    });
    expect(headingContainer).toHaveStyleRule('background', theme.colors.mainBg);
    expect(headingContainer).toMatchSnapshot();
  });
});
