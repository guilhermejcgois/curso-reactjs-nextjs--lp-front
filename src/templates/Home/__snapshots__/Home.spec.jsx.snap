// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render 1`] = `
.c0 {
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  z-index: 10;
  background-color: #0a1128;
}

.c0:before,
.c0:after {
  content: '';
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  border-width: 0.5rem;
  border-color: transparent;
  border-style: solid;
  border-radius: 50%;
}

.c0:after {
  width: 6rem;
  height: 6rem;
  border-left: 0.5rem solid #dc143c;
  border-top: 0.5rem solid #dc143c;
  -webkit-animation: cfKagt 600ms linear infinite;
  animation: cfKagt 600ms linear infinite;
}

.c0:before {
  width: 2rem;
  height: 2rem;
  border-left: 0.5rem solid #dc143c;
  border-top: 0.5rem solid #dc143c;
  -webkit-animation: cfKagt 1s linear reverse infinite;
  animation: cfKagt 1s linear reverse infinite;
}

<div>
  <div
    class="c0"
  />
</div>
`;
